src/uu/stty/src/stty.rs

```diff
@@ -210,31 +227,89 @@ fn stty(opts: &Options) -> UResult<()> {
+                            ),
+                            ControlCharMappingError::MultipleChars => {
+                                format!("invalid integer argument: '{mapping}'")
+                            }
+                        };
+                        USimpleError::new(1, message)
+                    })?;
+                    valid_args.push(ArgOptions::Mapping((char_index, cc_mapping)));
+                } else {
+                    return Err(USimpleError::new(1, format!("missing argument to '{arg}'")));
+                }
+            // non control char flag
+            } else if let Some(flag) = string_to_flag(arg) {
+                let mut remove_group = false;
+                match flag {
+                    AllFlags::Baud(_) => {}
+                    AllFlags::ControlFlags((flag, remove)) => {
+                        remove_group = check_flag_group(flag, remove);
+                    }
+                    AllFlags::InputFlags((flag, remove)) => {
+                        remove_group = check_flag_group(flag, remove);
+                    }
+                    AllFlags::LocalFlags((flag, remove)) => {
+                        remove_group = check_flag_group(flag, remove);
+                    }
+                    AllFlags::OutputFlags((flag, remove)) => {
+                        remove_group = check_flag_group(flag, remove);
+                    }
+                }
+                if remove_group {
+                    return Err(USimpleError::new(1, format!("invalid argument '{arg}'")));
+                }
+                valid_args.push(flag.into());
+            // not a valid control char or flag
+            } else {
+                return Err(USimpleError::new(1, format!("invalid argument '{arg}'")));
             }
         }
 
+        // TODO: Figure out the right error message for when tcgetattr fails
+        let mut termios = tcgetattr(opts.file.as_fd()).expect("Could not get terminal attributes");
+
+        // iterate over valid_args, match on the arg type, do the matching apply function
+        for arg in &valid_args {
+            match arg {
+                ArgOptions::Mapping(mapping) => apply_char_mapping(&mut termios, mapping),
+                ArgOptions::Flags(flag) => apply_setting(&mut termios, flag),
+            }
+        }
         tcsetattr(
             opts.file.as_fd(),
             nix::sys::termios::SetArg::TCSANOW,
             &termios,
         )
         .expect("Could not write terminal attributes");
     } else {
+        // TODO: Figure out the right error message for when tcgetattr fails
+        let termios = tcgetattr(opts.file.as_fd()).expect("Could not get terminal attributes");
         print_settings(&termios, opts).expect("TODO: make proper error here from nix error");
     }
     Ok(())
 }
 
+fn check_flag_group<T>(flag: &Flag<T>, remove: bool) -> bool {
+    if remove && flag.group.is_some() {
+        return true;
+    }
+    false
```

Comment:
You can simplify it to:
```suggestion
    remove && flag.group.is_some()
```